common/lib/xmodule/xmodule/js/src/video/09_video_caption.js

```diff
@@ -244,8 +244,26 @@ function (Sjson, AsyncProcess) {
                 success: function (sjson) {
                     self.sjson = new Sjson(sjson);
 
-                    var start = self.sjson.getStartTimes(),
-                        captions = self.sjson.getCaptions();
+                    // get start and caption. If startTime and endTime
+                    // are specified, filter by that range.
+                    var start,
+                        captions,
+                        startTime, 
+                        endTime,
+                        results,
+                        startTimeInSec = state.config.startTime,
+                        endTimeInSec = state.config.endTime;
+
+                    if (endTimeInSec !== null) {
```

Comment:
That's what I assume. The front-end translates `startTime` from the backend into either 0 or <= duration; either way, it's an integer. `endTimeInSec` can be defined as `null` if the backend reports an illegal value (see function `figureOutStartTime` here[https://github.com/Stanford-Online/edx-platform/blob/ataki/video-start-stop/common/lib/xmodule/xmodule/js/src/video/03_video_player.js#L734])